utilities/pipelines/resourceRemoval/helper/Invoke-ResourcePostRemoval.ps1

```diff
@@ -74,6 +74,31 @@ function Invoke-ResourcePostRemoval {
                 }
             }
         }
+        'Microsoft.OperationalInsights/workspaces' {
+            $subscriptionId = $resourceId.Split('/')[2]
+            $resourceGroupName = $resourceId.Split('/')[4]
+            $resourceName = Split-Path $ResourceId -Leaf
+            # Fetch service in soft-delete state
+            $getPath = '/subscriptions/{0}/providers/Microsoft.OperationalInsights/deletedWorkspaces?api-version=2020-03-01-preview' -f $subscriptionId
+            $getRequestInputObject = @{
+                Method = 'GET'
+                Path   = $getPath
+            }
+            $softDeletedService = ((Invoke-AzRestMethod @getRequestInputObject).Content | ConvertFrom-Json).value | Where-Object { $_.id -eq $resourceId }
+            if ($softDeletedService) {
+                # Recover service
+                $location = $softDeletedService.location
+                if ($PSCmdlet.ShouldProcess(('Log analytics workspace [{0}]' -f $resourceId), 'New')) {
+                    $recoveredWorkspace = New-AzOperationalInsightsWorkspace -ResourceGroupName $resourceGroupName -Name $resourceName -Location $location
+                }
+                if ($recoveredWorkspace) {
```

Comment:
Can this ever be empty?

Reply:
The only case would be if using `WhatIf` but even then it doesn't matter as the Purge also run's with WhatIf